doc/sphinx/language/core/variants.rst

```diff
@@ -24,6 +43,37 @@ Variants
    .. exn:: The @natural th argument of @ident must be @ident in @type.
       :undocumented:
 
+.. example::
+
+  The Booleans, the unit type and the empty types are respectively defined by:
+
+   .. coqtop:: in
+
+      Variant bool : Type := true : bool | false : bool.
+      Variant unit : Type := tt : unit.
+      Variant Empty_set : Type :=.
+
+  The option and sum types are defined by:
+
+   .. coqtop:: in
+
+      Variant option (A : Type) : Type := None : option A | Some : A -> option A.
+      Variant sum (A B : Type) : Type := inl : A -> sum A B | inr : B -> sum A B.
+
+  *Boolean reflection* is a relation reflecting under the form of a
+  Boolean value when a given proposition :n:`P` holds. It can be
+  defined as a two-constructor type family over :g:`bool`
+  parameterized by the proposition :n:`P`:
+
+  .. coqtop:: in
+
+     Variant reflect (P : Prop) : bool -> Set :=
+     | ReflectT : P -> reflect P true
+     | ReflectF : ~ P -> reflect P false.
+
+  :term:`Leibniz Equality` is another example of variant type.
```

Comment:
Actually, it's defined as a Inductive: `Inductive eq (A:Type) (x:A) : A -> Prop :=
    eq_refl : x = x :>A`

Reply:
The actual keyword that is used `Inductive` or `Variant` does not matter. This is a variant because it is not recursive. Maybe we should make this clearer in the doc...

Reply:
We are facing a dilemma here. It is indeed artificial to use `Inductive` for a type which is neither inductive nor coinductive but `Variant` does not define `eq_ind`/`eq_rect`, which are pervasively used.

So, we would need to add a "Set Nonrecursive Elimination Schemes" beforehand. But why not, if it helps to make thinks clearer.

Reply:
Probably worth mentioning the use of Inductive to create XX_ind and XX_rect as an alternative in the description of Variant.